src/solvers/augmented_Lagrangian_method.jl

```diff
@@ -42,8 +42,8 @@ manifold- or objective specific defaults.
 """
 mutable struct AugmentedLagrangianMethodState{
     P,
-    Pr<:AbstractManoptProblem,
-    St<:AbstractManoptSolverState,
+    Pr<:Union{F,AbstractManoptProblem} where {F},
+    St<:Union{AbstractEvaluationType,AbstractManoptSolverState},
```

Comment:
Are you sure this is right?

Reply:
See cases a) (first two types) and b) (second type) in the last comment. I think that is.